### PR TITLE
5.3 UnitTests

### PR DESCRIPTION
**Pull request by pi0**
I'm working on Unit Tests to be fully passed.

Update: All tests passed!

### CODE REVIEW

**Comment by pi0** [comment]
@jenssegers Finally all tests fully passed :) Would you please Review and Approve this PR ?

File: src/Jenssegers/Mongodb/Eloquent/Builder.php [hunk 1]
```diff
@@ -167,9 +168,10 @@ protected function addHasWhere(EloquentBuilder $hasQuery, Relation $relation, $o
         $query = $hasQuery->getQuery();
 
         // Get the number of related objects for each possible parent.
+        $relations = $query->pluck($relation->getHasCompareKey());
         $relationCount = array_count_values(array_map(function ($id) {
             return (string) $id; // Convert Back ObjectIds to Strings
-        }, $query->pluck($relation->getHasCompareKey())));
+        }, is_array($relations) ? $relations : $relations->toArray()));
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Eloquent/Builder.php]: This seems like a big change?

Comment (jenssegers) [inline src/Jenssegers/Mongodb/Eloquent/Builder.php]: Why was this needed?

**Comment by jenssegers** [comment]
Looking good!

File: src/Jenssegers/Mongodb/Eloquent/Model.php [hunk 2]
```diff
@@ -233,11 +234,11 @@ public function getAttribute($key)
 
                 // This attribute matches an embedsOne or embedsMany relation so we need
                 // to return the relation results instead of the interal attributes.
-                if ($relations instanceof EmbedsOneOrMany) {
+                if ($relations instanceof EmbedsOneOrMany or $relations instanceof MorphTo) {
```
Comment (jenssegers) [inline src/Jenssegers/Mongodb/Eloquent/Model.php]: This seems a bit odd, this is very specific to embedded relations, why was `MorpTo` added here?

File: src/Jenssegers/Mongodb/Eloquent/Builder.php (hunk 1 above)
Comment (pi0) [inline src/Jenssegers/Mongodb/Eloquent/Builder.php]: This tests fail without this check:
- RelationsTest.php:384
- RelationsTest.php:433

Error: `array_map(): Argument #2 should be an array`

Laravel `has` returns arrays instead of Collections!

File: src/Jenssegers/Mongodb/Eloquent/Model.php (hunk 2 above)
Comment (pi0) [inline src/Jenssegers/Mongodb/Eloquent/Model.php]: RelationsTest.php:367 : `Failed asserting that null is an instance of class "User".`
As a workaround to working MorphTo Relations will be directly returned and resolved.
Is it necessary to separate this logical part?

Comment (jenssegers) [inline src/Jenssegers/Mongodb/Eloquent/Model.php]: Might be better to split the logic indeed, I'll have to check what exactly is going wrong because this is weird.

**Comment by jenssegers** [comment]
Let me know when your changes are done!

**Comment by pi0** [comment]
@jenssegers last commit was due to accidental fast-forward :) i you think something is missing just let me know :)

**Comment by jenssegers** [comment]
Ill release this as an RC and go over everything soon.
